File: src/main/java/iesvdm/adrian/proyecto/domain/Equipo.java
package iesvdm.adrian.proyecto.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@EqualsAndHashCode(of = "id")
@Builder
@Table(name = "equipo")
public class Equipo implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_equipo")
    private Long id;
    @NotEmpty
    private String nombre;

    @Lob
    private String logo;

    @NotEmpty
    private String descripcion;

    @OneToMany(mappedBy = "equipo")
    private List<Usuario> jugadores;

    @JsonIgnore
    @ManyToMany(mappedBy = "equipos")
    private List<Torneo> torneosInscrito;
}

File: crud-front/src/app/model/equipo.ts
import { Partido } from "./partido";
import { Usuario } from "./usuario";
import { Torneo } from "./torneo";

export class Equipo {
    id!: number;
    nombre: string;
    descripcion: string;
    historial: Partido[];
    jugadores: Usuario[];
    torneosInstrito: Torneo[];

    constructor(nombre: string, url: string, descripcion: string, historial: Partido[], jugadores: Usuario[], torneosInstrito: Torneo[]){
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.historial = historial;
        this.jugadores = jugadores;
        this.torneosInstrito = torneosInstrito;
    }
}

